src/lib.rs

```diff
@@ -61,6 +61,18 @@ macro_rules! lsp_enum {
                 }
             }
         }
+
+        impl std::convert::TryFrom<&String> for $typ {
+            type Error = &'static str;
+            fn try_from(value: &String) -> Result<Self, Self::Error> {
+                match value {
+                    $(
+                        _ if *value == format!("{:?}", Self::$name) => Ok(Self::$name),
+                    )*
+                    _ => Err("unknown enum variant"),
+                }
+            }
+        }
```

Comment:
```suggestion
        impl std::convert::TryFrom<&str> for $typ {
            type Error = &'static str;
            fn try_from(value: &str) -> Result<Self, Self::Error> {
                match value {
                    $(
                        stringify!($name) => Ok(Self::$name),
                    )*
                    _ => Err("unknown enum variant"),
                }
            }
        }
```

Reply:
yeah, I was thinking to add a `&'static str` member to avoid the allocation.

Your suggestion is even more natural - though it requires `FUNCTION` as input instead of `Function`.
I don't have a strong opinion on which one is better. I guess we could keep symmetry with `Debug` while still only allocating once (converting `value`). Not sure what's better long-term.